### PR TITLE
Abstract <ListView/> Component

### PR DESCRIPTION
**Pull request by hueitan**
> props : 
> { items : { title, titleHtml, description, imageUrl }, containerRef }

### CODE REVIEW

File: src/components/ListView.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * The containerRef is suggested to be used together with the hooks useNavigation
+ * without the containerRef, the view won't scroll to the selected row
+ */
+export const ListView = ({ items, containerRef }) => {
+  if (!items.length) return
+  return <div class='results' ref={containerRef}>
+    {
+      items.map(item => (
+        <div class='result' data-selectable data-title={item.title} data-selected-key={item.title} key={item.title}>
```
Comment (stephanebisson) [inline src/components/ListView.js]: The assumed structure of item correspond to a search results item. Do you plan to factor out the details of `item` out of this component later or do you anticipate that this structure will actually work for all ListView usages?

File: style/listview.less [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+.result .img img {
+	width: 38px;
+	height: 38px;
+	border-radius: 4px;
+}
+
+.result[nav-selected="true"] {
+	background-color: #eaf3ff;
+}
```
Comment (stephanebisson) [inline style/listview.less]: Missing a line return here. 

I wish linting could catch that.

File: src/components/ListView.js (hunk 1 above)
Comment (hueitan) [reply src/components/ListView.js]: currently, what I foresee the common list view in KaiOS, items are all selectable, with the title, optional description, optional image.

I think we can do more changes when we use it on the other pages with different requirements.

File: style/listview.less (hunk 2 above)
Comment (hueitan) [reply style/listview.less]: not sure how many linting tools for less, maybe can try this https://github.com/lesshint/lesshint

out of scope now, can try doing this in other pr.

File: cypress/page-objects/search-page.js [hunk 3]
```diff
@@ -8,6 +8,6 @@ export class SearchPage {
     }
 
     results(){
-        return cy.get('.result')
+        return cy.get('.list')
```
Comment (stephanebisson) [inline cypress/page-objects/search-page.js]: This should probably be `.item`

**Comment by stephanebisson** [comment]
Something to think about for the future: can we encapsulate scrolling and positioning inside the ListView component.

**Comment by hueitan** [comment]
Yes, true, was thinking the same, the scrolling behavior should be handled in the listview component itself. will look into it.

**Pull request merged**